mongorestore/mongorestore_test.go

```diff
@@ -808,38 +820,59 @@ func TestFixDuplicatedLegacyIndexes(t *testing.T) {
 				}
 			}()
 
-			indexes := testDB.Collection("duplicate_index_key").Indexes()
-			c, err := indexes.List(context.Background())
+			c, err := testDB.Collection("duplicate_index_key").Indexes().List(context.Background())
 			So(err, ShouldBeNil)
+
 			type indexRes struct {
-				Key bson.D
+				Name string
+				Key  bson.D
 			}
 
-			var res indexRes
+			indexKeys := make(map[string]bson.D)
 
 			// two Indexes should be created in addition to the _id, foo and foo_2
-			c.Next(context.Background())
-			err = c.Decode(&res)
-			So(err, ShouldBeNil)
-			So(len(res.Key), ShouldEqual, 1)
-			So(res.Key[0].Key, ShouldEqual, "_id")
-			So(res.Key[0].Value, ShouldEqual, 1)
+			for c.Next(context.Background()) {
+				var res indexRes
+				err = c.Decode(&res)
+				So(err, ShouldBeNil)
+				So(len(res.Key), ShouldEqual, 1)
+				indexKeys[res.Name] = res.Key
+			}
 
-			c.Next(context.Background())
-			err = c.Decode(&res)
-			So(err, ShouldBeNil)
-			So(len(res.Key), ShouldEqual, 1)
-			So(res.Key[0].Key, ShouldEqual, "foo")
-			So(res.Key[0].Value, ShouldEqual, 1)
+			So(len(indexKeys), ShouldEqual, 3)
 
-			c.Next(context.Background())
-			err = c.Decode(&res)
-			So(err, ShouldBeNil)
-			So(len(res.Key), ShouldEqual, 1)
-			So(res.Key[0].Key, ShouldEqual, "foo")
-			So(res.Key[0].Value, ShouldEqual, 2)
+			var indexKey bson.D
+			// Check that only one of foo_, foo_1, or foo_1.0 was created
+			indexKeyFoo, ok := indexKeys["foo_"]
+			indexKeyFoo1, ok1 := indexKeys["foo_1"]
+			indexKeyFoo10, ok10 := indexKeys["foo_1.0"]
 
-			So(c.TryNext(context.Background()), ShouldBeFalse)
+			So(ok || ok1 || ok10, ShouldBeTrue)
```

Comment:
I had to redo this test since the index catalog uses a map and the order in which indexes are built is now random.